### ISSUE TITLE
Exception serializing an anonymous object

### ISSUE THREAD

**Issue by pedrompc** [ch-robinson/dotnet-avro#246]
I am attempting to perform serialization on an anonymous object using the SerializerBuilder, so that I can produce using a byte[], however I am getting an exception. 

Here's the code I'm using to serialize:
```
using var serializerBuilder = new SchemaRegistrySerializerBuilder(schemaRegistry);
var keySerializationContext = new SerializationContext(MessageComponentType.Key, request.TopicTarget);
var keySerializer = await serializerBuilder.Build<object>(9);
var keybytes = keySerializer.Serialize(new { Key = "key1"}, keySerializationContext);
```

The schema, registered under id = 9
```
{
  "type": "record",
  "name": "DefaultKeyEntity",
  "namespace": "Test",
  "fields": [
    {
      "name": "Key",
      "type": "string"
    }
  ]
}
```

I am getting the following exception:

```
Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: Cannot perform runtime binding on a null reference
   at CallSite.Target(Closure , CallSite , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)
   at DefaultKeyEntity serializer(Closure , Object , BinaryWriter )
   at Chr.Avro.Confluent.DelegateSerializer`1.Serialize(T data, SerializationContext context)
```

In case this type of usage is not supported by the library, I was wondering if you had any tips on the best way to produce messages to a topic, without having a strongly typed class to start from. I basically need to serialize a dynamic object using a schema. 
I have tried dynamically constructing a GenericRecord from a dynamic, using the Apache.Avro library, however I have run into some limitations when trying to build nested records.

Thank you

**Comment by dstelljes** [comment]
Thanks for the report; this is a case that the library should support but presently does not. We’ll investigate it as a bug and provide updates here.

As a workaround, you should be able to accomplish this with dynamic objects:

```csharp
dynamic key = new ExpandoObject();
key.Key = "key1";
keySerializer.Serialize(key, keySerializationContext);
```

**Comment by dstelljes** [comment]
Apologies for the wait on this. The issue turned out to be two separate bugs, one in how the Schema Registry serializer was being built and another in how properties were being accessed. We'll release both fixes shortly with 9.4.1.
